stdlib/ballerina-mime/src/main/java/org/ballerinalang/mime/util/MimeUtil.java

```diff
@@ -358,12 +351,16 @@ public static BXML getXmlPayload(BStruct entity) {
         if (byteData != null) {
             return entity.getBlobField(BYTE_DATA_INDEX);
         } else {
-            BStruct fileHandler = (BStruct) entity.getRefField(OVERFLOW_DATA_INDEX);
-            String filePath = fileHandler.getStringField(FILE_PATH_INDEX);
-            return readFromFile(filePath);
+            String filePath = getFilePathFromFileHandler(entity);
+            return filePath.isEmpty() ? null : readFromFile(filePath);
         }
     }
 
+    private static String getFilePathFromFileHandler(BStruct entity) {
```

Comment:
Again, is `FileHandler` the correct term here, for the `File` struct?

Reply:
Addressed with b98823c.